uast/uast.go

```diff
@@ -216,7 +220,11 @@ func RoleList(roles ...role.Role) nodes.Array {
 }
 
 // RolesOf is a helper for getting node UAST roles (see KeyRoles).
-func RolesOf(m nodes.Object) role.Roles {
+func RolesOf(n nodes.Node) role.Roles {
+	m, ok := n.(nodes.Object)
```

Comment:
As 185. Is there some reason for this relaxation more generally?

Reply:
The only reason to relax this requirement is that all XPath query and any manual node manipulations will usually return a `nodes.Node`, and the user have to type assert it each time to use those functions.